src/main.rs

```diff
@@ -62,8 +67,14 @@ async fn main() -> eyre::Result<()> {
                 }),
                 _ => None,
             };
+
+            if let Some(addr) = address {
+                info!("Using {} for transmission", addr);
+            }
+
             ServiceData::Transmission(Transmission {
                 rpc_credentials,
+                address: address.clone(),
```

Comment:
We might stop pattern matching on references here, since we do not need the `args.service` after the `match` itself, thus eliminating the need for this `.clone()`.